src/YoutubeAtom/YoutubeOverlay.tsx

```diff
@@ -1,52 +1,188 @@
+    const [isPlayerReady, setIsPlayerReady] = useState(false);
+    const [player, setPlayer] = useState<YT.Player | null>(null);
+
+    const loadVideo = () => {
+        setPlayer(
+            new window.YT.Player(`${id}`, {
+                events: {
+                    onReady: () => setIsPlayerReady(true),
+                },
+                playerVars: {
+                    mute: 1,
+                    autoplay: 1,
+                },
+            }),
+        );
+    };
+
+    useEffect(() => {
+        if (window.YT) {
+            loadVideo();
+        } else {
+            // If not, load the script asynchronously
+            const tag = document.createElement('script');
+            tag.src = 'https://www.youtube.com/iframe_api';
+
+            // onYouTubeIframeAPIReady will load the video after the script is loaded
+            window.onYouTubeIframeAPIReady = loadVideo;
+
+            const firstScriptTag = document.getElementsByTagName('script')[0];
+            firstScriptTag &&
+                firstScriptTag.parentNode &&
+                firstScriptTag.parentNode.insertBefore(tag, firstScriptTag);
+        }
+    }, []);
+
+    const onClickOverlay = useCallback(() => {
+        if (isPlayerReady && player) {
+            try {
+                player.playVideo();
+                setHideOverlay(true);
+            } catch (e) {
+                console.error(`Unable to play video due to: ${e}`);
+            }
+        }
+    }, [player, isPlayerReady]);
+
+    const onKeydownOverlay = useCallback(
+        (e) => {
+            const spaceKey = 32;
+            const enterKey = 13;
+            if (
+                (e.keyCode === spaceKey || e.keyCode === enterKey) &&
+                isPlayerReady &&
+                player
+            ) {
+                try {
+                    player.playVideo();
+                    setHideOverlay(true);
+                } catch (e) {
+                    console.error(`Unable to play video due to: ${e}`);
+                }
+            }
+        },
+        [player, isPlayerReady],
+    );
+
+    return (
+        <div
+            // cannot use cx because it would cause new key to be generated on `hideOverlay` toggle causing css refresh
+            className={`${overlayStyles(image)} ${
+                hideOverlay ? hideOverlayStyling : ''
+            }`}
+            onClick={onClickOverlay}
+            onKeyDown={onKeydownOverlay}
+            tabIndex={0}
+        >
+            <div className={overlayInfoWrapperStyles}>
+                <div className={`${playButtonStyling} overlay-play-button`}>
+                    <SvgPlay />
+                </div>
+                {duration && (
+                    <div className={videoDurationStyles}>
+                        {formatTime(duration)}
+                    </div>
+                )}
+            </div>
             <YoutubeMeta mediaDuration={duration} pillar={pillar} />
```

Comment:
Might need to remove this if we are formatting and adding the time in the overlay class